src/compiler/builderState.ts

```diff
@@ -48,6 +48,11 @@ namespace ts {
          */
         readonly exportedModulesMap: BuilderState.ManyToManyPathMap | undefined;
 
+        previousCache?: {
+            id: number,

```

Comment:
I think that if you swap this to the containing `ReadonlyManyToManyPathMap`, you can avoid introducting an `id` field to `ReadonlyManyToManyPathMap` itself.

I don't know if it adds any risk of introducing a leak of some sort, but it might be work considering.

Reply:
I was worried about the hypothetical leak.  Plus, the ID was useful to tracing/debugging.